types/i18next/index.d.ts

```diff
@@ -1,157 +1,711 @@
+        fallbackNS?: false | string | string[];
+
+        /**
+         * calls save missing key function on backend if key not found
+         * @default false
+         */
         saveMissing?: boolean;
-        saveMissingTo?: string;
-        missingKeyHandler?: (lng: string, ns: string, key: string, fallbackValue: string) => void;
-        parseMissingKeyHandler?: (key: string) => void;
+
+        /**
+         * 'current' or 'all'
+         * @default fallback
+         */
+        saveMissingTo?: "current" | "all" | "fallback";
+
+        /**
+         * function(lng, ns, key, fallbackValue) { } used for custom missing key handling (needs saveMissing set to true!)
+         * @default false
+         */
+        missingKeyHandler?: false | ((lng: string, ns: string, key: string, fallbackValue: string) => void);
+
+        /**
+         * function(key) { // return value to display }
+         * @default noop
+         */
+        parseMissingKeyHandler?(key: string): string;
+
+        /**
+         * appends namespace to missing key
+         * @default false
+         */
         appendNamespaceToMissingKey?: boolean;
-        postProcess?: string | any[];
+
+        /**
+         * will use 'plural' as suffix for languages only having 1 plural form, setting it to false will suffix all with numbers
+         * @default true
+         */
+        simplifyPluralSuffix?: boolean;
+
+        /**
+         * string or array of postProcessors to apply per default
+         * @default false
+         */
+        postProcess?: false | string | string[];
+
+        /**
+         * allows null values as valid translation
+         * @default true
+         */
         returnNull?: boolean;
+
+        /**
+         * allows empty string as valid translation
+         * @default true
+         */
         returnEmptyString?: boolean;
+
+        /**
+         * allows objects as valid translation result
+         * @default false
+         */
         returnObjects?: boolean;
-        returnedObjectHandler?: (key: string, value: string, options: any) => void;
+
+        /**
+         * function(key, value, options) {} gets called if object was passed in as key but returnObjects was set to false
+         * @default noop
+         */
+        returnedObjectHandler?(key: string, value: string, options: any): void;
+
+        /**
+         * char, eg. '\n' that arrays will be joined by
+         * @default false
+         */
+        joinArrays?: false | string;
+
+        /**
+         * default: sets defaultValue
+         * @default "(args) => ({ defaultValue: args[1] })"
+         */
+        overloadTranslationOptionHandler?(args: string[]): TranslationOptions;
+
+        /**
+         * see interpolation
+         * @default {...}
+         */
+        interpolation?: InterpolationOptions;
+
+        /**
+         * options for language detection - check documentation of plugin
+         * @default undefined
+         */
+        detection?: {};
```

Comment:
Yes, I did not notice that definitions have TypeScript 2.3 version. We should use `object` type here and below for `backend` and `cache`.